### ISSUE TITLE
Specific name of the current chain is not displayed

### ISSUE THREAD

**Issue by AI-Chef** [hwchase17/langchain#7524]
### System Info

LangChain v0.0.229, Python v3.10.12, Ubuntu 20.04.2 LTS

### Who can help?

@hwchase17 @agola11 

### Information

- [ ] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [X] Chains
- [X] Callbacks/Tracing
- [ ] Async

### Reproduction

I am encountering an issue where the specific name of the current chain is not being displayed in the console output, even though I have set 'verbose=True' in the MultiPromptChain and other Chains. When the program enters a new chain, it only prints 'Entering new  chain...' without specifying the name of the chain. This makes it difficult to debug and understand which chain is currently being used. Could you please look into this issue and provide a way to display the name of the current chain in the console output? Thank you.

The output could be
```
> Entering new  chain...


> Entering new  chain...
lib/python3.10/site-packages/langchain/chains/llm.py:275: UserWarning: The predict_and_parse method is deprecated, instead pass an output parser directly to LLMChain.
  warnings.warn(

> Finished chain.
math: {'input': 'What is the derivative of a function?'}

> Entering new  chain...
Prompt after formatting:
You are a very good mathematician. You are great at answering math questions. \nYou are so good because you are able to break down hard problems into their component parts, \nanswer the component parts, and then put them together to answer the broader question.
Here is a question:
What is the derivative of a function?

> Finished chain.

> Finished chain.
```

### Expected behavior

```
> Entering new MultiPromptChain chain...


> Entering new LLMRouterChain chain...
lib/python3.10/site-packages/langchain/chains/llm.py:275: UserWarning: The predict_and_parse method is deprecated, instead pass an output parser directly to LLMChain.
  warnings.warn(

> Finished chain.
math: {'input': 'What is the derivative of a function?'}

> Entering new LLMChain[math] chain...
Prompt after formatting:
You are a very good mathematician. You are great at answering math questions. \nYou are so good because you are able to break down hard problems into their component parts, \nanswer the component parts, and then put them together to answer the broader question.
Here is a question:
What is the derivative of a function?

> Finished chain.

> Finished chain.
```

**Comment by AI-Chef** [comment]
My code is followed the documentation https://python.langchain.com/docs/modules/chains/foundational/router

**Comment by gaurang98671** [comment]
Hello @AI-Chef,  I think its because of on_chain_start() method is not able to parse the chain name in [stdout.py](https://github.com/hwchase17/langchain/blob/6cdd4b5edca511b0015f1b39102225fe638d8359/langchain/callbacks/stdout.py) file, as there is no "name" field in "serialized" dictionary.
I did some debugging and found out that there is an "id" field in that dictionary which is a list. Last element in that list is the name of chain class that you are looking for.
```
{'lc': 1, 'type': 'not_implemented', 'id': ['langchain', 'chains', 'router', 'multi_prompt', 'MultiPromptChain']}
``` 

Solution
```
 def on_chain_start(
        self, serialized: Dict[str, Any], inputs: Dict[str, Any], **kwargs: Any
    ) -> None:
        """Print out that we are entering a chain."""
        class_name = serialized.get("id", "")[-1]
        print(f"\n\n\033[1m> Entering new {class_name} chain...\033[0m")
```
Final output
```
←[1m> Entering new MultiPromptChain chain...←[0m
{'lc': 1, 'type': 'not_implemented', 'id': ['langchain', 'chains', 'router', 'llm_router', 'LLMRouterChain']}


←[1m> Entering new LLMRouterChain chain...←[0m
^c
```
I hope this helps :)

**Comment by gaurang98671** [comment]
@AI-Chef, ahh mybad, thanks for pointing that out.

**Comment by AI-Chef** [comment]
Solved in pull #6124
